web/directory/views.py

```diff
@@ -1,434 +1,318 @@
-        serializer = CoopSpreadsheetSerializer(coops, many=True)
-        return Response(serializer.data)
+            queryset = queryset.filter(filter)
+        
+        public_addresses = CoopAddressTags.objects.filter(is_public=True)
+        public_contactmethods = ContactMethod.objects.filter(is_public=True)
+        public_people = Person.objects.filter(is_public=True)
 
-    def post(self, request, format=None):
-        serializer = CoopSpreadsheetSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        people_with_public_contact_methods = Prefetch('contact_methods', queryset=public_contactmethods)
+        public_people = Person.objects.prefetch_related(people_with_public_contact_methods).filter(is_public=True)
 
-class CoopDetail(APIView):
-    """
-    Retrieve, update or delete a coop instance.
-    """
-    def get_object(self, pk):
-        try:
-            return Coop.objects.get(pk=pk)
-        except Coop.DoesNotExist:
-            raise Http404
+        queryset = queryset.prefetch_related(
+            Prefetch('addresses', queryset=public_addresses),
+            Prefetch('contact_methods', queryset=public_contactmethods),
+            Prefetch('people', queryset=public_people),
+        )
 
-    def get(self, request, pk, format=None):
-        coop = self.get_object(pk)
-        serializer = CoopSerializer(coop)
-        return Response(serializer.data)
+        return queryset
+    
+    def get_permissions(self):
+        if self.request.method == 'GET': #LIST
+            self.permission_classes = [AllowAny]
+        elif self.request.method == 'POST': #CREATE
+            self.permission_classes = [IsAuthenticated]
+        return [permission() for permission in self.permission_classes]
+    
+class CoopDetail(generics.RetrieveAPIView):
+    queryset = Coop.objects.filter(status=Coop.Status.ACTIVE)
+    serializer_class = CoopSerializer
+    permission_classes = [AllowAny]
 
-    def put(self, request, pk, format=None):
-        coop = self.get_object(pk)
-        serializer = CoopSerializer(coop, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    def get_object(self):
+        coop_public_id = self.kwargs.get('coop_public_id')
 
-    def patch(self, request, pk, format=None):
-        coop = self.get_object(pk)
-        serializer = CoopProposedChangeSerializer(coop, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        public_contact_methods = ContactMethod.objects.filter(is_public=True)
```

Comment:
Let's talk this over with Steve - we might want to have phone and emails separated out in this view, unless Steve is okay with lumping them together (e.g. in the spreadsheet page)